src/runtime_src/core/pcie/windows/core_system.cpp

```diff
@@ -32,11 +33,77 @@ void xrt_core::system::get_xrt_info(boost::property_tree::ptree &_pt)
 }
 
 
-void xrt_core::system::get_os_info(boost::property_tree::ptree &_pt)
+//function for converting CString type to std::string
+std::string convert(CString input)
 {
-  // TODO
-  _pt.put("windows", "");
+  CT2CA converted_input(input);
+  std::string string_input(converted_input);
+
+  return string_input;
 }
 
 
+CString GetStringFromReg(HKEY keyParent, CString keyName, CString keyValName)
+{
+  CRegKey key;
+  CString out;
+
+  if (key.Open(keyParent, keyName, KEY_READ) == ERROR_SUCCESS) {
+	ULONG len = 256;
+	key.QueryStringValue(keyValName, out.GetBuffer(256), &len);
+	out.ReleaseBuffer();
+	key.Close();
+  }
+
+  return out;
+}
+
+
+CString getmachinename()
+{
+  CString machine;
+  SYSTEM_INFO sysInfo;
 
+  // Get hardware info
+  ZeroMemory(&sysInfo, sizeof(SYSTEM_INFO));
+  GetSystemInfo(&sysInfo);
+  // Set processor architecture
+  switch (sysInfo.wProcessorArchitecture) {
+  case PROCESSOR_ARCHITECTURE_AMD64:
+	  machine = CString(_T("x86_64"));
+	  break;
+  case PROCESSOR_ARCHITECTURE_IA64:
+	  machine = CString(_T("ia64"));
+	  break;
+  case PROCESSOR_ARCHITECTURE_INTEL:
+	  machine = CString(_T("x86"));
+	  break;
+  case PROCESSOR_ARCHITECTURE_UNKNOWN:
+  default:
+	  machine = CString(_T("unknown"));
+	  break;
+  }
+
+  return machine;
+}
+
+
+void xrt_core::system::get_os_info(boost::property_tree::ptree &_pt)
+{
+  CString osversion = GetStringFromReg(HKEY_LOCAL_MACHINE,
+		L"SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion", L"CurrentVersion");
+  CString osname = GetStringFromReg(HKEY_LOCAL_MACHINE,
+		L"SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion", L"ProductName");
+  CString build = GetStringFromReg(HKEY_LOCAL_MACHINE,
+		L"SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion", L"BuildLab");
+  CString machine = getmachinename();
+  char tnow[26];
+  time_t result = time(NULL);
+
+  ctime_s(tnow, sizeof tnow, &result);
+  _pt.put("sysname", convert(osname));
```

Comment:
Please refactor the code to be something like this:
_pt.put("sysname", GetStringFromReg(HKEY_LOCAL_MACHINE, "SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion", "ProductName").c_str());

Question: Does the underlying calls support the following variable name?
"SOFTWARE/Microsoft/Windows NT/CurrentVersion"